avm/res/dev-test-lab/lab/main.bicep

```
@@ -314,7 +315,7 @@ output resourceId string = lab.id
 output name string = lab.name
 
 @description('The principal ID of the system assigned identity.')
-output systemAssignedMIPrincipalId string = contains(lab.identity, 'principalId') ? lab.identity.principalId : ''
+output systemAssignedMIPrincipalId string = lab.identity.principalId
```

```suggestion
output systemAssignedMIPrincipalId string = lab.?identity.?principalId
```

Given that the value is seemingly optional, please consider the above suggestion.

---

@AlexanderSehr system identity will always be returned. I may have misunderstood your suggestion but what does it achieve in this case?

---

Pardon, I was looking at the input and thoughts it's optional, forgetting that the resource would then just create an MSI. It that weren't the case, the syntax would ensure we'd try to access the identity value without failing if it doesn't exist.